### PR TITLE
parser: implement Restore for TraceStmt

### PR DESCRIPTION
**Pull request by GunaYX**
- implement Restore for `TraceStmt`

pingcap/tidb#8532

### CODE REVIEW

**Comment by leoppro** [comment]
Thanks for your contribution.

File: go.mod1 [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/cznic/sortutil v0.0.0-20150617083342-4c7342852e65
 	github.com/cznic/strutil v0.0.0-20171016134553-529a34b1c186
 	github.com/cznic/y v0.0.0-20170802143616-045f81c6662a
-	github.com/pingcap/check v0.0.0-20181213055612-5c2b07721bdb
+	github.com/pingcap/check v0.0.0-20190102082844-67f458068fc8
```
Comment (leoppro) [inline go.mod1]: revert go.mod1 and go.sum1 files

File: ast/misc.go [hunk 2]
```diff
@@ -101,7 +101,15 @@ type TraceStmt struct {
 
 // Restore implements Node interface.
 func (n *TraceStmt) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("TRACE ")
```
Comment (leoppro) [inline ast/misc.go]: we should add test here:
https://github.com/pingcap/parser/blob/7b0cfe281af20fb68fe83bd5d66c03c6dcfb6d9d/parser_test.go#L2297-L2304

but we need the implement of `dmlstmt`s to test `TraceStmt`

Comment (leoppro) [reply ast/misc.go]: all of `dmlstmt`s were implemented, you can test this pr now.

**Comment by GunaYX** [comment]
Added test cases, the tasked is finished now.
@leoppro

File: parser_test.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/pingcap/parser/mysql"
 	"github.com/pingcap/parser/terror"
 	"github.com/pingcap/tidb/types"
-	"github.com/pingcap/tidb/types/parser_driver"
+	driver "github.com/pingcap/tidb/types/parser_driver"
```
Comment (kennytm) [inline parser_test.go]: I don't think this line needs to be changed.

File: ast/misc.go [hunk 4]
```diff
@@ -115,7 +115,16 @@ type TraceStmt struct {
...
+	if n.Format != "json" {
+		ctx.WriteKeyWord("FORMAT = ")
```
Comment (kennytm) [inline ast/misc.go]:
```suggestion
		ctx.WriteKeyWord("FORMAT")
		ctx.WritePlain(" = ")
```

**Comment by tiancaiamao** [comment]
Ping @GunaYX

**Comment by GunaYX** [comment]
@kennytm suggestions implemented

**Pull request merged**